### PR TITLE
switching from $.livechange to _.debounce

### PR DESCRIPTION
**Pull request by 2is10**
@joonho1101 ok with you? underscore.js is pretty well known, documented, and tested. I can already think of another place we ought to be using `_.debounce`, and I plan to make the switch there too. This will mean one less custom jQuery plugin for first-time readers of our codebase to learn and for us to maintain.

### CODE REVIEW

File: packrat/scripts/tagbox.js [hunk 1]
```diff
@@ -192,13 +192,13 @@ this.tagbox = (function ($, win) {
...
-				var text = e.value;
-				text = text.trim();
-
-				this.$inputbox.toggleClass('empty', !text);
-
-				this.suggest(text);
+			function onInput(e) {
+				var text = e.target.value.trim();
+				if (text !== this.text) {
+					this.text = text;
```
Comment (2is10) [inline packrat/scripts/tagbox.js:206]: I’d be open to other suggestions about where to stash `text` if you’d prefer not using `this`.

**Comment by andrewconner** [comment]
Related: http://lodash.com/ is an (almost) drop in replacement for underscore that benchmarks much better: http://jsperf.com/lo-dash-v1-1-1-vs-underscore-v1-4-4/2

**Comment by 2is10** [comment]
That benchmark is for iterating over an array, and we aren’t using that part of underscore. We use native for loops or the native `Array.prototype.forEach`. Although, I must say, it’s quite impressive that lodash beats native `Array.prototype.forEach` (slightly) on most browsers.

**Comment by andrewconner** [comment]
From what I've seen, it's faster for most functions. lodash's goal is efficiency, so makes some decisions that underscore wouldn't merge in (for instance from above, it uses a faster-than-native-forEach looping method, and underscore decided to always fall back to forEach if it exists).

**Comment by 2is10** [comment]
I’ve compared the source for these two methods, `throttle` and `debounce`. lodash supports extra options. Its code is larger and slower. underscore’s is simpler and doing the bare minimum.

If/when we add more underscore methods, I promise to evaluate the corresponding lodash implementations first. Happy? :smile:

**Comment by andrewconner** [comment]
Always :)

**Pull request merged**